File: 242-valid-anagram/242-valid-anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        map<char,int> characters;
        if(s.size()!=t.size())
            return false;
        for(char ch : s)
        {
            if(characters.find(ch)!=characters.end())
            {
                characters[ch]++;
            }
            else
            {
                characters.insert(pair<char,int>(ch,1));
            }
        }
        for(char ch : t)
        {
            if(characters.find(ch)!=characters.end() && characters[ch]!=0)
            {
                characters[ch]--;        
            }
            else
                return false;
                  
        }
        return true;
    }
};

File: 242-valid-anagram/242-valid-anagram.java
//MY SOLUTION USING HASHMAPS
class Solution {
    public boolean isAnagram(String s, String t) {
        HashMap<Character,Integer> sCharSet = new HashMap<>();
        HashMap<Character,Integer> tCharSet = new HashMap<>();
        
        if(s.length()!=t.length())
            return false;
        
        for(int i=0;i<t.length();i++)
        {
            char ch = s.charAt(i);
            sCharSet.put(ch, sCharSet.getOrDefault(ch,0)+1);
            char chr = t.charAt(i);
            tCharSet.put(chr, tCharSet.getOrDefault(chr,0)+1);
        }
        
        for(Map.Entry<Character,Integer> mp : sCharSet.entrySet())
        {
            if(!tCharSet.containsKey(mp.getKey()) || tCharSet.get(mp.getKey()).intValue()!=mp.getValue().intValue())
            {
                return false;
            }
        }
        return true;
    } 
}

// class Solution {
//     public boolean isAnagram(String s, String t) {
        
//     }
// }